src/security/ServerOptions.cc

```diff
@@ -50,7 +54,18 @@ Security::ServerOptions::parse(const char *token)
     }
 
     // parse the server-only options
-    if (strncmp(token, "dh=", 3) == 0) {
+    if (strncmp(token, "context=", 8) == 0) {
+#if USE_OPENSSL
+        staticContextSessionId = SBuf(token+8);
+        if (staticContextSessionId.length() > SSL_MAX_SSL_SESSION_ID_LENGTH) {
+            debugs(83, DBG_CRITICAL, "FATAL: Option 'context=' value is too long. Maximum " << SSL_MAX_SSL_SESSION_ID_LENGTH << " characters.");
+            self_destruct();
+        }
+#else
+        debugs(83, DBG_PARSE_NOTE(DBG_IMPORTANT), "WARNING: Option 'context=' requires --with-openssl. Ignoring.");
```

Comment:
> If an anyone can figure out what the value is they have better ability to reuse the sessions in other places that use the same value.

I do not think knowing a constant (session context ID) inside an encrypted value (session state) makes it any easier to reuse that session with malicious intent.

> Since we are offering it as arbitrary configurable string to admin we cannot know for sure whether they are using it for weak origin-scoping like Tomcat, or for more critical uses like scoping TLS authentication within an enterprise.

If knowing the value does not weaken security, then (for this discussion) it does not matter what the value is used for.

> I'm failing to see how this would detect a stale session cache entry unless one was regularly changing the string value ... Using it with normal values of host:port will not detect the age/staleness of any particular session cache entry - only the origin it was created for.

I believe you have pretty much addressed your own doubts -- the context ID can be used to detect stale values if and only if the ID changes when the old context becomes stale. What "stale" means depends on the deployment, of course. In some cases, it could be a change in server host name or port. In some other cases, it could be a date change. In yet other cases, it could be the version change of some monitoring software. The actual staleness definition does not matter for this discussion.